src/components/map/MapService.js

```diff
@@ -371,12 +371,51 @@ goog.require('ga_urlutils_service');
    */
   module.provider('gaLayers', function() {
 
-    this.$get = function($http, $q, $rootScope, $translate, $window,
+    this.$get = function($http, $q, $rootScope, $translate, $window, $timeout,
         gaBrowserSniffer, gaDefinePropertiesForLayer, gaMapUtils,
         gaNetworkStatus, gaStorage, gaTileGrid, gaUrlUtils,
         gaStylesFromLiterals, gaGlobalOptions, gaPermalink, 
         gaLang, gaTime) {
 
+      var MAX_RETRIES = 100;
+      var RETRIES_INTERVAL = 5000;
+
+      var retryOnError = function(msg, el, olsource) {
+        if (angular.isDefined(msg) &&
+            angular.isDefined(olsource)) {
```

Comment:
Warning `angular.isDefined(null)` returns true.  I'm a fan of using ` if(msg && source)` instead.